src/components/Modal/__stories__/modal.stories.mdx

```diff
@@ -0,0 +1,384 @@
+import Modal from "../Modal";
+import { useModalHelper } from "./helpers";
+import { ArgsTable, Story, Canvas, Meta } from "@storybook/addon-docs";
+import { createStoryMetaSettings } from "../../../storybook";
+import "./Modal.stories.scss";
+import { Button, ModalContent, ModalFooter, ModalHeader, TextField } from "components";
+import {
+  TOOLTIP,
+  ALERT_BANNER,
+  Tipseen
+} from "../../../storybook/components/related-components/component-description-map";
+import { Settings } from "components/Icon/Icons";
+import { RelatedComponents } from "../../../storybook"; import { useState } from "react";
+
+export const metaSettings = createStoryMetaSettings({
+  component: Modal,
+  enumPropNamesArray: [], // List enum props here
+  iconPropNamesArray: [], // List props that are typed as icons here
+  actionPropsArray: [] // List the component's actions here
+});
+
+<Meta
+  title="Feedback/Modal"
+  component={Modal}
+  subcomponents={{ ModalHeader, ModalContent, ModalFooter }}
+  argTypes={metaSettings.argTypes}
+  decorators={metaSettings.decorators}
+/>
+
+<!--- Component documentation -->
+
+# Modal
+
+- [Overview](#overview)
+- [Props](#props)
+- [Usage](#usage)
+- [Variants](#variants)
+- [Do’s and don’ts](#dos-and-donts)
+- [Use cases and examples](#use-cases-and-examples)
+- [Related components](#related-components)
+- [Feedback](#feedback)
+
+## Overview
+
+An easy to use accessible Modal component
+
+<Canvas>
+  <Story name="Overview">
+    {() => {
```

Comment:
The reason that the controls are not working in this story is because you don't use the component template syntax. Please see how we implement the overview story in other components (for example button) as a reference